tripy/tests/integration/conftest.py

```diff
@@ -0,0 +1,86 @@
+#
+# SPDX-FileCopyrightText: Copyright (c) 2024-2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pytest
+import inspect
+
+import tripy as tp
+
+
+@pytest.fixture(params=["compile", "eager"])
+def execution_mode(request):
+    return request.param
+
+
+@pytest.fixture
+def compile_fixture(execution_mode):
+    def wrapper(func, *args, exclude_arg_names=None, **kwargs):
+        def get_shape(x: tp.Tensor):
+            x.eval()
+            return tp.InputInfo(x.trace_tensor.shape, dtype=x.dtype)
+
+        exclude_arg_names = set() if exclude_arg_names is None else set(exclude_arg_names)
+
+        if execution_mode == "compile":
```

Comment:
nit: we can do an early exit here and unindent the rest of the code:
```suggestion
        if execution_mode == "eager":
            return func(*args, **kwrags) 
        
        assert execution_mode == "compile"
        ...
```